content/frontend/react-apollo/8-subscriptions.md

```diff
@@ -1,201 +1,225 @@
+is used to "route" a request to a specific middleware link.
+It takes three arguments, the first one is a `test` function
+which returns a boolean. The remaining two arguments are
+again of type `ApolloLink`. If `test` returns `true`, the
+request will be forwarded to the link passed as the second
+argument. If `false`, to the third one.
+
+In our case, the `test` function is checking whether the
+requested operation is a _subscription_. If it is, it will
+be forwarded to the `wsLink`, otherwise (if it's a _query_
+or _mutation_), the `authLink.concat(httpLink)` will take
+care of it:
 
 ![](https://cdn-images-1.medium.com/max/720/1*KwnMO21k0d3UbyKWnlbeJg.png)
-_Picture taken from [Apollo Link: The modular GraphQL network stack](https://dev-blog.apollodata.com/apollo-link-the-modular-graphql-network-stack-3b6d5fcf9244) by [Evans Hauser](https://twitter.com/EvansHauser)_
-
-### Subscribing to new links
-
-For the app to update in realtime when new links are created, you need to subscribe to events that are happening on the `Link` type. There generally are three kinds of events you can subscribe to when using Prisma:
+_Picture taken from
+[Apollo Link: The modular GraphQL network stack](https://dev-blog.apollodata.com/apollo-link-the-modular-graphql-network-stack-3b6d5fcf9244)
+by [Evans Hauser](https://twitter.com/EvansHauser)_
 
-- a new `Link` is _created_
-- an existing `Link` is _updated_
-- an existing `Link` is _deleted_
+### Subscribing to New links
 
-You'll implement the subscription in the `LinkList` component since that's where all the links are rendered.
+For the app to update in realtime when new links are
+created, we need to subscribe to events that are happening
+on the `Link` type. We'll implement the subscription in the
+`LinkList` component since that's where all the links are
+rendered.
 
 <Instruction>
 
-Open `LinkList.js` and update current component as follow:
+The `useQuery` hook provided by Apollo gives us access to a
+function called `subscribeToMore`. We can destructure this
+function out and use it to act on new data that comes in
+over a subscription. This will give us the effect of making
+our app "realtime".
 
-```js{11-13,18,22}(path=".../hackernews-react-apollo/src/components/LinkList.js")
-class LinkList extends Component {
-  _updateCacheAfterVote = (store, createVote, linkId) => {
-    const data = store.readQuery({ query: FEED_QUERY })
+Open `LinkList.js` and update current component as follow:
 
-    const votedLink = data.feed.links.find(link => link.id === linkId)
-    votedLink.votes = createVote.link.votes
+```js{}(path=".../hackernews-react-apollo/src/components/LinkList.js")
+const LinkList = () => {
+  // ...
 
-    store.writeQuery({ query: FEED_QUERY, data })
-  }
+  const {
+    data,
+    loading,
+    error,
+    subscribeToMore
+  } = useQuery(FEED_QUERY, {
+    variables: getQueryVariables(isNewPage, page)
```

Comment:
`getQueryVariables`, `isNewPage` and  `page` are not defined. Kindly provide the default values